baybe/surrogates/random_forest.py

```diff
@@ -65,38 +55,46 @@ def _make_target_scaler_factory() -> type[OutcomeTransform] | None:
         # Tree-like models do not require any output scaling
         return None
 
-    @batchify
-    def _estimate_moments(
-        self, candidates_comp_scaled: Tensor, /
-    ) -> tuple[Tensor, Tensor]:
+    def _posterior(self, candidates_comp_scaled: Tensor, /) -> EnsemblePosterior:
         # See base class.
 
+        from botorch.models.ensemble import EnsemblePosterior
+
         # FIXME[typing]: It seems there is currently no better way to inform the type
         #   checker that the attribute is available at the time of the function call
         assert self._model is not None
 
-        import torch
+        @batchify_ensemble_predictor
+        def predict(candidates_comp_scaled: Tensor) -> Tensor:
+            """Make the end-to-end ensemble prediction."""
+            import torch
 
-        # Evaluate all trees
-        # NOTE: explicit conversion to ndarray is needed due to a pytorch issue:
-        # https://github.com/pytorch/pytorch/pull/51731
-        # https://github.com/pytorch/pytorch/issues/13918
-        predictions = torch.from_numpy(
-            np.asarray(
-                [
-                    self._model.estimators_[tree].predict(candidates_comp_scaled)
-                    for tree in range(self._model.n_estimators)
-                ]
+            return torch.from_numpy(
+                self._predict_ensemble(
+                    self._model.estimators_, candidates_comp_scaled.numpy()
+                )
             )
-        )
 
-        # Compute posterior mean and variance
-        mean = predictions.mean(dim=0)
-        var = predictions.var(dim=0)
+        return EnsemblePosterior(predict(candidates_comp_scaled).unsqueeze(-1))
+
+    @staticmethod
+    def _predict_ensemble(
+        predictors: Collection[Callable[[np.ndarray], np.ndarray]],
+        candidates: np.ndarray,
+    ) -> Tensor:
+        """Evaluate an ensemble of predictors on a given candidate set."""
+        # Extract shapes
+        n_candidates = len(candidates)
+        n_estimators = len(predictors)
+
+        # Evaluate all trees
+        predictions = np.zeros((n_estimators, n_candidates))
+        for p, predictor in enumerate(predictors):
+            predictions[p] = predictor.predict(candidates)
 
-        return mean, var
+        return predictions
```

Comment:
type hint says `Tensor` but this is numpy

Reply:
Generally forgot to check mypy (there were also a couple of other issues, yours included). Now fixed